tests/test_models/test_tbats.py

```diff
@@ -126,3 +162,157 @@ def test_prediction_interval(model, example_tsds):
+@pytest.mark.parametrize(
+    "estimator",
+    (
+        BATSModel,
+        TBATSModel,
+    ),
+)
+def test_predict_components_not_implemented(periodic_ts, estimator):
+    train, test = periodic_ts
+    model = estimator()
+    model.fit(train)
+
+    with pytest.raises(NotImplementedError, match="Prediction decomposition isn't currently implemented!"):
+        for segment in test.columns.get_level_values("segment"):
+            model._models[segment].predict_components(horizon=3)
+
+
+@pytest.mark.long_2
+@pytest.mark.parametrize(
+    "estimator",
+    (
+        BATSModel,
+        TBATSModel,
+    ),
+)
+def test_decompose_forecast_output_format(periodic_ts, estimator):
+    horizon = 5
+    train, test = periodic_ts
+    model = estimator()
+    model.fit(train)
+
+    for segment in test.columns.get_level_values("segment"):
+        components = model._models[segment]._decompose_forecast(horizon=horizon)
+        assert isinstance(components, np.ndarray)
+        assert components.shape[0] == horizon
+
+
+@pytest.mark.long_2
+@pytest.mark.parametrize(
+    "estimator",
+    (
+        BATSModel,
+        TBATSModel,
+    ),
+)
+def test_named_components_output_format(periodic_ts, estimator):
+    horizon = 5
+    train, test = periodic_ts
+    model = estimator()
+    model.fit(train)
+
+    for segment in test.columns.get_level_values("segment"):
+        components = model._models[segment]._decompose_forecast(horizon=horizon)
+        named_components = model._models[segment]._named_components(raw_components=components)
+
+        for component in named_components.values():
+            assert len(component) == horizon
+
+
+@pytest.mark.long_2
+@pytest.mark.parametrize(
+    "estimator",
+    (
+        BATSModel,
+        TBATSModel,
+    ),
+)
+@pytest.mark.parametrize(
+    "data,params,message",
+    (
+        (
+            "sinusoid_ts",
+            {"use_box_cox": False, "use_trend": False, "use_arma_errors": False, "seasonal_periods": [7, 14]},
+            "Seasonal components is not fitted!",
+        ),
+        (
+            "periodic_ts",
+            {
+                "use_box_cox": True,
+                "use_trend": True,
+                "use_damped_trend": True,
+                "use_arma_errors": True,
+                "seasonal_periods": [7, 14],
+            },
+            "ARMA components is not fitted!",
+        ),
+    ),
+)
+def test_fitted_components_warnings(data, estimator, params, message, request):
+    train, test = request.getfixturevalue(data)
+
+    model = estimator(**params)
+    model.fit(train)
+
+    with pytest.warns(Warning, match=message):
+        for segment in test.columns.get_level_values("segment"):
+            model._models[segment].forecast_components(horizon=3)
+
+
+@pytest.mark.long_2
+@pytest.mark.filterwarnings("ignore:.* is not fitted!$")
+@pytest.mark.parametrize(
+    "estimator",
+    (
+        BATSModel,
+        TBATSModel,
+    ),
+)
+@pytest.mark.parametrize(
```

Comment:
May be better to do the parametrization as cartesian product like
```
@pytest.mark.parametrize("use_box_cox", [True, False])
@pytest.mark.parametrize("use_trend", [True, False])
...
```

Reply:
Right now test cases are cherry-picked. If we do cartesian product it will increase test cases to at least 16 per model instead of 7. This will significantly slow down testing process.

Reply:
Actually, it doesn't seem right. For trend/arma/seasonality (seasonality on/off) there are 8 combinations. Dampen trend should be included as well as it used in grid search and impact the trend component. So the number of combinations should be multiplied by 2 once again.